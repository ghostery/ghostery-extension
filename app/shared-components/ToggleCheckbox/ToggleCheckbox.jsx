/**
 * Toggle Checkbox Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

/**
 * A Functional React component for a Toggle Checkbox
 * @return {JSX} JSX for rendering a Toggle Checkbox
 * @memberof SharedComponents
 */
const ToggleCheckbox = (props) => {
	const checkboxClassNames = ClassNames('ToggleCheckbox', {
		'ToggleCheckbox--active': props.checked,
	});

	return (
		<div className={checkboxClassNames} onClick={props.onChange}>
			<svg viewBox="0 0 24 24">
				{props.checked ? (
					<path d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z" />
				) : (
					<path d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z" />
				)}
			</svg>
		</div>
	);
};

// PropTypes ensure we pass required props of the correct type
ToggleCheckbox.propTypes = {
	checked: PropTypes.bool.isRequired,
	onChange: PropTypes.func.isRequired,
};

export default ToggleCheckbox;
