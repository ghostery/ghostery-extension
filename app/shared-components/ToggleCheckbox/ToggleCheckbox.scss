/**
 * Toggle Checkbox Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2019 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Toggle Checkbox
.ToggleCheckbox {
	height: 48px;
	width: 48px;
	padding: 12px;
	cursor: pointer;
	flex-shrink: 0;
}
.ToggleCheckbox path {
	fill: #6d6d6d;
}
.ToggleCheckbox.ToggleCheckbox--active path {
	fill: #1dafed;
}
.ToggleCheckbox--flush-left {
	width: 36px;
	padding-left: 0px;
}
