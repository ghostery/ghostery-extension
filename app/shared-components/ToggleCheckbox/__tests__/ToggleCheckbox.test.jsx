/**
 * Toggle Checkbox Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ToggleCheckbox from '../ToggleCheckbox';

describe('app/shared-components/ToggleCheckbox component', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('toggle checkbox is rendered correctly when checked', () => {
			const initialState = {
				checked: true,
				onChange: () => {},
			};

			const component = renderer.create(
				<ToggleCheckbox {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('toggle checkbox is rendered correctly when not checked', () => {
			const initialState = {
				checked: false,
				onChange: () => {},
			};

			const component = renderer.create(
				<ToggleCheckbox {...initialState} />
			).toJSON();
			expect(component).toMatchSnapshot();
		});

	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				checked: false,
				onChange: jest.fn(),
			};

			const component = shallow(<ToggleCheckbox {...initialState} />);
			expect(component.find('.ToggleCheckbox').length).toBe(1);
			expect(component.find('.ToggleCheckbox svg').length).toBe(1);
			expect(component.find('.ToggleCheckbox svg path').length).toBe(1);
			expect(component.find('.ToggleCheckbox.ToggleCheckbox--active').length).toBe(0);

			expect(initialState.onChange.mock.calls.length).toBe(0);
			component.find('.ToggleCheckbox').simulate('click');
			expect(initialState.onChange.mock.calls.length).toBe(1);

			expect(component.find('.ToggleCheckbox--active').length).toBe(0);
			component.setProps({ checked: true });
			expect(component.find('.ToggleCheckbox--active').length).toBe(1);
		});
	});
});
