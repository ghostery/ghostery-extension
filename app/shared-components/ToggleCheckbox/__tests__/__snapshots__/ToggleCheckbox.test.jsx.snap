// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/ToggleCheckbox component Snapshot tests with react-test-renderer toggle checkbox is rendered correctly when checked 1`] = `
<div
  className="ToggleCheckbox ToggleCheckbox--active"
  onClick={[Function]}
>
  <svg
    viewBox="0 0 24 24"
  >
    <path
      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
</div>
`;

exports[`app/shared-components/ToggleCheckbox component Snapshot tests with react-test-renderer toggle checkbox is rendered correctly when not checked 1`] = `
<div
  className="ToggleCheckbox"
  onClick={[Function]}
>
  <svg
    viewBox="0 0 24 24"
  >
    <path
      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
    />
  </svg>
</div>
`;
